docs/en-us/dev/user_doc/alert_spi.md

```diff
@@ -0,0 +1,102 @@
+
+When you need to extend, you actually only need to focus on the extension interface. For example, for the alert service you only need to focus on the AlertChannelFactory and the AlertChannel. The underlying logic is already implemented by DolphinScheduler, which makes our development much more focused and simple.
+
+By the way, we use an excellent front-end component, form-create, which supports the generating of front-end ui components based on json. If plugin development involves a front-end, we will use json to generate the relevant front-end UI components. The parameters of the plugin are wrapped in org.apache.dolphinscheduler.spi.params, which converts all the relevant parameters into json, meaning that you can draw the front-end components (mainly forms here, we only care about the data for front and back-end interaction) in Java code.
+
+This article focuses on the design and development of Alert alerts.
+
+#### Main Modules
+
+If you don't care about its internal design, just want to know how to develop your own alarm plugin, you can skip this section.
+
+* dolphinscheduler-spi
+
+     This module is the core SPI module and provides the basic SPI relevant behaviour,where dolphinschedulerplugin is the plugin top layer interface, all DolphinsCheduler's plugins must implement the interface, and the module also provides some universal tool classes (if you can detach it again Some will something will be better? For example, the UI, we currently use the parameter this block) and some UI related basic information.
+
+* dolphinscheduler-alert
+
+     In this module, we have implemented the load of the associated plugin when the Alert-Server starts. Alert provides a variety of plug-in configuration methods that can be enabled by simple configurations after you have done a job. The configuration file is located at dolphinscheduler-alert/src/main/resources/alert.properties .
+
+     It provides two methods of configuration.
+
+     1: Configure the jar directory specified by the plugin, eg: alert.plugin.dir=/root/dolphinscheduler/lib/plugin/alert . When alert-server starts, it will load the jar of the relevant plugin from the specified directory.
+
+     2: IDE development mode
+
+     You can use this configuration when you are in the debugging phase of development, see [dolphinscheduler-maven-plugin](https://github.com/apache/incubator-dolphinscheduler-maven-plugin) for design principles.
+
+     
+
+ * 3: Packaging plugins
+
+  For plugin packaging we use [provisio](https://github.com/jvanzyl/provisio), which is an excellent packaging tool that you need to add to dolphinscheduler-dist/src/main/provisio/ after you have finished developing the plugin. dolphinscheduler.xml, which will package the plugin to the specified directory when it is executed.

```

Comment:
For plugin packaging we use [provisio](https://github.com/jvanzyl/provisio), which is an excellent packaging tool that you need to add to dolphinscheduler-dist/src/main/provisio/ after you have finished developing the plugin. dolphinscheduler.xml, which will package the plugin to the specified directory when it is executed. =>
We use [provisio](https://github.com/jvanzyl/provisio), an excellent packaging tool, for packaging plugins. Please add it to dolphinscheduler-dist/src/main/provisio/dolphinscheduler.xml, and it will package plugins to the specified directory when processed.